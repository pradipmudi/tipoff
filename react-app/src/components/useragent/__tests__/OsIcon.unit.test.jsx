import React from "react";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TestOsIcon from '../OsIcon';

describe("OsIcon", () => {

    it("matches snapshot", () => {
        const os = {
            name : 'Windows',
            version : '10'
        }
        const tree = renderer.create(<TestOsIcon os={os} />).toJSON();
        expect(tree).toMatchSnapshot();
    
    });
    
    it("renders without issues", () =>{
        const os = {
            name : 'Windows',
            version : '10'
        }
        const component = shallow(<TestOsIcon os={os} />);
        expect(component.find('.fab').hasClass('fa-windows')).toEqual(true);
        expect(component.find('.fab').props().title)
            .toEqual(`${os.name}:${os.version}`);
    });

    it("renders with default icon when browser is not passed", () => {
        const component = shallow(<TestOsIcon  />);
        expect(component.find('.far').hasClass('fa-window-maximize')).toEqual(true);
    })

});
