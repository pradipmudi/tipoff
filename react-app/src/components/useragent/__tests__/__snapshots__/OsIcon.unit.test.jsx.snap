// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OsIcon matches snapshot 1`] = `
<span>
  <span
    className="icon"
    title="Windows:10"
  >
    <i
      aria-hidden="true"
      className="fab fa-windows"
    />
  </span>
</span>
`;
